src/test/run-pass/issues/issue-54696.rs

```diff
@@ -0,0 +1,8 @@
+// run-pass
+
+fn main() {
+    // We shouldn't promote this
+    &(main as fn() == main as fn());
+    // Also check nested case
+    &(&(main as fn()) == &(main as fn()));
+}
```

Comment:
How does this test verify lack of promotion? Shouldn't we instead have a compile-fail test?

Reply:
The promotion causes crashes, because function pointer equality can't be done in constants